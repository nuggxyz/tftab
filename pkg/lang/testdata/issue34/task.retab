TOOLS_DIR = "tools"

bins = {
	"task"          = "github.com/go-task/task/v3/cmd/task"
	"buf"           = "github.com/bufbuild/buf/cmd/buf",
	"mockery"       = "github.com/vektra/mockery/v2",
	"gotestsum"     = "gotest.tools/gotestsum",
	"golangci-lint" = "github.com/golangci/golangci-lint/cmd/golangci-lint",
}

gen taskfile {
	// schema = "https://taskfile.dev/schema.json"
	path = "../taskfile.yaml"
	data = {
		version = 3
		tasks = {
				for bin, path in bins : "${bin}-bin" => {
					generates = ["bin/${bin}"]
					dir       = "${TOOLS_DIR}"
					sources   = ["go.mod", "go.sum", "main.go"]
					cmds = [
						"go build -mod=vendor -o ./bin/${bin} ${path}"
					]
				}
			}
	}
}
