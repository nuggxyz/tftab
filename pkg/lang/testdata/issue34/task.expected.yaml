# code generated by retab. DO NOT EDIT.
# join the fight against yaml @ github.com/walteh/retab

# source: "task.retab"

version: 3
tasks:
    task-bin:
        generates:
            - bin/task
        dir: tools
        sources:
            - go.mod
            - go.sum
            - main.go
        cmds:
            - go build -mod=vendor -o ./bin/task github.com/go-task/task/v3/cmd/task
    buf-bin:
        generates:
            - bin/buf
        dir: tools
        sources:
            - go.mod
            - go.sum
            - main.go
        cmds:
            - go build -mod=vendor -o ./bin/buf github.com/bufbuild/buf/cmd/buf
    mockery-bin:
        generates:
            - bin/mockery
        dir: tools
        sources:
            - go.mod
            - go.sum
            - main.go
        cmds:
            - go build -mod=vendor -o ./bin/mockery github.com/vektra/mockery/v2
    gotestsum-bin:
        generates:
            - bin/gotestsum
        dir: tools
        sources:
            - go.mod
            - go.sum
            - main.go
        cmds:
            - go build -mod=vendor -o ./bin/gotestsum gotest.tools/gotestsum
    golangci-lint-bin:
        generates:
            - bin/golangci-lint
        dir: tools
        sources:
            - go.mod
            - go.sum
            - main.go
        cmds:
            - go build -mod=vendor -o ./bin/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
